homeassistant/components/ecobee/sensor.py

```diff
@@ -19,22 +19,49 @@
 
 SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
     SensorEntityDescription(
-        key="temperature",
-        name="Temperature",
-        native_unit_of_measurement=TEMP_FAHRENHEIT,
-        device_class=SensorDeviceClass.TEMPERATURE,
-        state_class=SensorStateClass.MEASUREMENT,
+        key = "temperature",
+        name = "Temperature",
+        native_unit_of_measurement = TEMP_FAHRENHEIT,
+        device_class = SensorDeviceClass.TEMPERATURE,
+        state_class = SensorStateClass.MEASUREMENT,
     ),
     SensorEntityDescription(
-        key="humidity",
-        name="Humidity",
-        native_unit_of_measurement=PERCENTAGE,
-        device_class=SensorDeviceClass.HUMIDITY,
-        state_class=SensorStateClass.MEASUREMENT,
+        key = "humidity",
+        name = "Humidity",
+        native_unit_of_measurement = PERCENTAGE,
+        device_class = SensorDeviceClass.HUMIDITY,
+        state_class = SensorStateClass.MEASUREMENT,
     ),
+    SensorEntityDescription(
+        key = "co2PPM",
+        name = "CO2",
+        native_unit_of_measurement = CONCENTRATION_PARTS_PER_MILLION,
+        device_class = SensorDeviceClass.CO2,
+        state_class = SensorStateClass.MEASUREMENT,
+     ),
+    SensorEntityDescription(
+        key = "vocPPM",
+        name = "VOC",
+        device_class = SensorDeviceClass.VOLATILE_ORGANIC_COMPOUNDS,
+        native_unit_of_measurement = CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class = SensorStateClass.MEASUREMENT,
+     ),
+    SensorEntityDescription(
+        key = "airQuality",
+        name = "Air Quality Index",
+        device_class = SensorDeviceClass.AQI,
+        native_unit_of_measurement = None,
+        state_class = SensorStateClass.MEASUREMENT
+     )
 )
 
-
+runtimeKeys = {
```

Comment:
Probably want to reformat to match codestyle

```python
runtimeKeys = {
   "temperature": 'actualTemperature',
   "humidity": 'actualHumidity',
   "co2PPM": 'actualCO2',
   "vocPPM": 'actualVOC',
   "airQuality": 'actualAQScore'
}
```

Reply:
Fixed

Reply:
Please store the runtime value key in an attribute on each entity description instead.

Use a mixin dataclass to add a required attribute to the sensor description.

Example:
https://github.com/home-assistant/core/blob/74cfdc6c1fa1e9fb4fa7acb6b4d06401f7ae66b6/homeassistant/components/wled/sensor.py#L34-L45

Reply:
I’m not sure I entirely understand. 

I would define the mixin class which would just contain a string attribute, and put the “actual…” values in that attribute when the SENSOR tuple gets defined?

Reply:
Yes, we create the sensor entity descriptions as constants in `SENSOR_TYPES`.

Reply:
So I would create a mixin class with a single attribute:

runtimeKey: str 

then at the SENSOR tuple define it as derived from SensorDescriptionEntity, and the mixin, and just set that runtimeKey there?

Reply:
Yes, exactly. Please name the attribute `runtime_key`. We follow PEP8 and use snake_case variable names.

Reply:
Can you give me a less complex example?

I'm trying this, and am trying to figure out what's wrong:
                                                                               
@dataclass                                                                      
class EcobeeSensorEntityDescriptionMixin:                                       
    runtime_key: str                                                            
                                                                                
                                                                                
@dataclass                                                                      
class EcobeeSensorEntityDescription(                                            
    SensorEntityDescription, EcobeeSensorEntityDescriptionMixin                 
):                                                                              
                                                                                
                                                                                
                                                                                
SENSOR_TYPES: tuple[EcobeeSensorDescription, ...] = (                           
    EcobeeSensorEntityDescription(                                              
        key="temperature",                                                      
        name="Temperature",                                                     
        native_unit_of_measurement=TEMP_FAHRENHEIT,                             
        device_class=SensorDeviceClass.TEMPERATURE,                             
        state_class=SensorStateClass.MEASUREMENT,                               
        runtime_key="actualTemperature",                                        
    ),                                                                          
    EcobeeSensorEntityDescription(         

[etc...]

Reply:
nvm, need a pass statement for empty class.